src/main.rs

```diff
@@ -150,7 +153,30 @@ mod tests {
         // Create output directory if it doesn't exist
         fs::create_dir_all("./test_data/output").unwrap();
 
-        let price_provider = GoferPriceProvider::new("cat ./test_data/input.json");
+        let mut server = mockito::Server::new();
```

Comment:
Very nice, definetely cleaner than my `cat testdata` "mock". 
Feel free to delete the `test_data/input.json` since it does not seem to be used anywhere else I think.